pkg/services/extsvcauth/registry/service.go

```diff
@@ -17,32 +18,76 @@ type Registry struct {
 	logger      log.Logger
 	oauthServer oauthserver.OAuth2Server
 	saSvc       *extsvcaccounts.ExtSvcAccountsService
+
+	extSvcProvider map[string]extsvcauth.AuthProvider
+	lock           sync.Mutex
 }
 
 func ProvideExtSvcRegistry(oauthServer oauthserver.OAuth2Server, saSvc *extsvcaccounts.ExtSvcAccountsService, features featuremgmt.FeatureToggles) *Registry {
 	return &Registry{
-		features:    features,
-		logger:      log.New("extsvcauth.registry"),
-		oauthServer: oauthServer,
-		saSvc:       saSvc,
+		extSvcProvider: map[string]extsvcauth.AuthProvider{},
+		features:       features,
+		lock:           sync.Mutex{},
+		logger:         log.New("extsvcauth.registry"),
+		oauthServer:    oauthServer,
+		saSvc:          saSvc,
+	}
+}
+
+// HasExternalService returns whether an external service has been saved with that name.
+func (r *Registry) HasExternalService(ctx context.Context, name string) bool {
+	_, ok := r.extSvcProvider[name]
```

Comment:
This technically just checks that we have received a registration. IMO this is good enough, worst case, calling remove will be a no-op.